pkg/volume/plugins.go

```diff
@@ -133,6 +133,18 @@ type AttachableVolumePlugin interface {
 	VolumePlugin
 	NewAttacher() (Attacher, error)
 	NewDetacher() (Detacher, error)
+
+	// GetUniqueVolumeName returns a unique name representing the volume
+	// defined in spec. e.g. pluginname-deviceName-readwrite
+	// This helps ensures that the same operation (attach/detach) is never
+	// started on the same volume.
+	// If the plugin does not support the given spec, this returns an error.
+	GetUniqueVolumeName(spec *Spec) (string, error)
+
+	// GetDiskName returns the name of the device referenced in the specified
+	// spec. This is passed by callers to the Deatch method.
+	// If the plugin does not support the given spec, this returns an error.
+	GetDeviceName(spec *Spec) (string, error)
```

Comment:
optional nit: this should be GetDeviceID. AWS for example does not allow you to give names to volumes and you have to use its ID.